test/e2e/apm/association_test.go

```diff
@@ -43,6 +45,41 @@ func TestCrossNSAssociation(t *testing.T) {
 	test.Sequence(nil, test.EmptySteps, esBuilder, apmBuilder).RunSequential(t)
 }
 
+// TestAPMKibanaAssociation tests associating an APM Server with Kibana.
+func TestAPMKibanaAssociation(t *testing.T) {
+	stackVersion := version.MustParse(test.Ctx().ElasticStackVersion)
+	if !stackVersion.IsSameOrAfter(apmv1.ApmAgentConfigurationMinVersion) {
+		t.SkipNow()
+	}
+
+	ns := test.Ctx().ManagedNamespace(0)
+	name := "test-apm-kb-assoc"
+
+	esBuilder := elasticsearch.NewBuilder(name).
+		WithNamespace(ns).
+		WithESMasterDataNodes(1, elasticsearch.DefaultResources).
+		WithRestrictedSecurityContext()
+
+	kbBuilder := kibana.NewBuilder(name).
+		WithNamespace(ns).
+		WithElasticsearchRef(esBuilder.Ref()).
+		WithNodeCount(1).
+		WithRestrictedSecurityContext()
+
+	apmBuilder := apmserver.NewBuilder(name).
+		WithNamespace(ns).
+		WithElasticsearchRef(esBuilder.Ref()).
+		WithKibanaRef(kbBuilder.Ref()).
+		WithNodeCount(1).
+		WithRestrictedSecurityContext().
+		WithConfig(map[string]interface{}{
+			"apm-server.ilm.enabled":                           false,
+			"setup.template.settings.index.number_of_replicas": 0, // avoid ES yellow state on a 1 node ES cluster
```

Comment:
:o I'm surprised APM does not rely on `auto_expand_replicas: "0-1"` similar to what Kibana is doing.

Reply:
Looks like they considered it at some point https://github.com/elastic/apm-server/pull/503